gnocchi/carbonara.py

```diff
@@ -85,6 +84,46 @@ def round_timestamp(ts, freq):
         (ts - UNIX_UNIVERSAL_START64) / freq) * freq
 
 
+def make_timeseries_array(array):
+    """Create a Numpy array with the right format for a timeseries."""
+    return numpy.array(array,
+                       dtype=[('timestamps', 'datetime64[ns]'),
+                              ('values', 'float64')])
+
+
+def make_timeseries(timestamps, values):
+    """Return a Numpy array representing a timeseries.
+
+    This array specifies correctly the data types, which is important for
+    Numpy to operate fastly.
+    """
+    l = len(timestamps)
+    if l != len(values):
+        raise ValueError("Timestamps and values must have the same length")
+    arr = make_timeseries_array(numpy.zeros(l))
+    arr['timestamps'] = timestamps
+    arr['values'] = values
+    return arr
+
+
+def combine_timeseries(ts1, ts2):
```

Comment:
i think we should be good:

```
In [48]: [blah,blah2]
Out[48]: 
[array([('2017-07-27T13:33:21.000000000',  1.),
        ('2017-07-27T13:34:27.000000000',  2.)],
       dtype=[('timestamps', '<M8[ns]'), ('values', '<f8')]),
 array([('2017-07-27T13:33:27.000000000',  3.)],
       dtype=[('timestamps', '<M8[ns]'), ('values', '<f8')])]

In [49]: %timeit __, index = np.unique(np.append(blah['timestamps'], blah2['timestamps']), return_index=True); np.append(blah, blah2)[
    ...: index]
The slowest run took 8.65 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 22.9 µs per loop
```

vs

```
In [56]: ts = pd.Series(blah['values'], blah['timestamps'])

In [57]: ts2 = pd.Series(blah2['values'], blah2['timestamps'])

In [58]: %timeit ts.combine_first(ts2)
1000 loops, best of 3: 619 µs per loop
```

too lazy to make something with real size